src/pages/library/DataBrowserPreview.js

```diff
@@ -0,0 +1,109 @@
+import _ from 'lodash/fp'
+import { useState } from 'react'
+import { div, h, h1 } from 'react-hyperscript-helpers'
+import { Select } from 'src/components/common'
+import FooterWrapper from 'src/components/FooterWrapper'
+import { centeredSpinner, spinner } from 'src/components/icons'
+import { libraryTopMatter } from 'src/components/library-common'
+import { SimpleTable } from 'src/components/table'
+import { Ajax } from 'src/libs/ajax'
+import colors from 'src/libs/colors'
+import { withErrorReporting } from 'src/libs/error'
+import * as Utils from 'src/libs/utils'
+import { useDataCatalog } from 'src/pages/library/dataBrowser-utils'
+
+
+const styles = {
+  table: {
+    header: {
+      color: colors.accent(),
+      height: '1rem',
+      textTransform: 'uppercase', fontWeight: 600, fontSize: '0.75rem'
+    }
+  }
+}
+
+const activeTab = 'browse & explore'
+let metadata
+let tableMap
+
+const selectTable = ({ id, value, signal, setSelectedTable, setPreviewData, loading, setLoading }) => {
+  const loadTable = _.flow(
```

Comment:
Could this be moved to the `useOnMount` below? I think it may simplify the code a bit.

Reply:
i moved the whole selectTable block into the DataBrowserPreview method - the selectTable method needs to be called in the autocomplete block too, so I just want that method to be reuseable. If I'm misunderstanding, please let me know!